util/packaging/homebrew/chapel-main.rb

```diff
@@ -56,6 +57,11 @@ def install
     (libexec/"chplconfig").write <<~EOS
       CHPL_RE2=bundled
       CHPL_GMP=system
+      CHPL_MEM=jemalloc
+      CHPL_HOST_MEM=jemalloc
```

Comment:
Note that `CHPL_HOST_MEM` defaults to cstdlib on mac. Not sure we need/want to set it here. 

Also don't see any reason to explicitly set `CHPL_MEM=jemalloc`

Reply:
I added the `CHPL_HOST_MEM` line because without it I was running into this (confusing) error message:

`Error: CHPL_JEMALLOC must be 'none' when CHPL_MEM is not jemalloc`

Confusing because this is despite the fact that I explicitly did set `CHPL_MEM` to `jemalloc`.

Taking a look at `chpl_jemalloc.get`: 
https://github.com/chapel-lang/chapel/blob/dbf58e62045702a039cc9a6332b061632fdcd1c9/util/chplenv/chpl_jemalloc.py#L11

it can be called with its `flag` parameter as either `host` or `target`. If it's `host` then the `mem_val` on this line is really referring to `CHPL_HOST_MEM` and `jemalloc_val` is really referring to `CHPL_HOST_JEMALLOC`:

https://github.com/chapel-lang/chapel/blob/dbf58e62045702a039cc9a6332b061632fdcd1c9/util/chplenv/chpl_jemalloc.py#L57-L58

Maybe we should have the error print `CHPL_HOST_JEMALLOC`/`CHPL_TARGET_JEMALLOC` and `CHPL_HOST_MEM`/`CHPL_TARGET_MEM` mem as appropriate. Though that might be confusing if the user really did set `CHPL_MEM` directly and [this line in our doc](https://chapel-lang.org/docs/main/usingchapel/chplenv.html#chpl-mem) makes me thing maybe it's going to be a moot point in a release or two anyway:

> CHPL_TARGET_MEM will be replacing CHPL_MEM in the future. CHPL_TARGET_MEM takes precedence over CHPL_MEM.

In terms of what to do for this homebrew formula: now that I write this out, it seems like a better fix than setting `CHPL_HOST_MEM=jemalloc` would be to remove the `CHPL_HOST_JEMALLOC=system` line (in which case I think it defaults to none?). Trying it out this seems to build fine.


> Also don't see any reason to explicitly set CHPL_MEM=jemalloc

Yep, looks like that one's not needed.

Reply:
> In terms of what to do for this homebrew formula: now that I write this out, it seems like a better fix than setting `CHPL_HOST_MEM=jemalloc` would be to remove the `CHPL_HOST_JEMALLOC=system` line (in which case I think it defaults to none?). Trying it out this seems to build fine.

👍